Source/Core/DolphinQt2/Settings/GameCubePane.cpp

```diff
@@ -0,0 +1,333 @@
+
+  // Memcard validity checks
+  if (memcard)
+  {
+    GCMemcard mc(filename.toStdString());
+
+    if (!mc.IsValid())
+    {
+      QMessageBox::critical(this, tr("Error"), tr("Cannot use that file as a memory card.\n%s\n"
+                                                  "is not a valid GameCube memory card file")
+                                                   .arg(filename));
+      return;
+    }
+
+    QString path_b =
+        QFileInfo(QString::fromStdString(slot == 0 ? SConfig::GetInstance().m_strMemoryCardB :
+                                                     SConfig::GetInstance().m_strMemoryCardA))
+            .absolutePath();
+
+    if (path_abs == path_b)
+    {
+      QMessageBox::critical(this, tr("Error"),
+                            tr("Are you trying to use the same file in both slots?"));
+      return;
+    }
+  }
+
+  if (memcard)
+  {
+    if (slot == 0)
+    {
+      SConfig::GetInstance().m_strMemoryCardA = path_abs.toStdString();
+    }
+    else
+    {
+      SConfig::GetInstance().m_strMemoryCardB = path_abs.toStdString();
+    }
+  }
+  else
+  {
+    if (slot == 0)
+    {
+      SConfig::GetInstance().m_strGbaCartA = path_abs.toStdString();
+    }
+    else
+    {
+      SConfig::GetInstance().m_strGbaCartB = path_abs.toStdString();
+    }
+  }
+}
+
+void GameCubePane::LoadSettings()
+{
+  const SConfig& params = SConfig::GetInstance();
+
+  // IPL Settings
+  m_skip_main_menu->setChecked(params.bHLE_BS2);
+  m_language_combo->setCurrentIndex(params.SelectedLanguage);
+  m_override_language_ntsc->setChecked(params.bOverrideGCLanguage);
+
+  bool have_menu = false;
+
+  for (const std::string& dir : {USA_DIR, JAP_DIR, EUR_DIR})
+  {
+    const auto& path = DIR_SEP + dir + DIR_SEP GC_IPL;
+    if (File::Exists(File::GetUserPath(D_GCUSER_IDX) + path) ||
+        File::Exists(File::GetUserPath(D_GCUSER_IDX) + path))
+    {
+      have_menu = true;
+      break;
+    }
+  }
+
+  m_skip_main_menu->setEnabled(have_menu);
+  m_skip_main_menu->setToolTip(have_menu ? QStringLiteral("") :
+                                           tr("Put Main Menu roms in User/GC/{region}."));
+
+  // Device Settings
+
+  for (int i = 0; i < 3; i++)
+  {
+    int index = 0;
+    switch (SConfig::GetInstance().m_EXIDevice[i])
+    {
+    case ExpansionInterface::EXIDEVICE_NONE:
+      index = 0;
+      break;
+    case ExpansionInterface::EXIDEVICE_DUMMY:
+      index = 1;
+      break;
+    case ExpansionInterface::EXIDEVICE_MEMORYCARD:
+      index = 2;
+      break;
+    case ExpansionInterface::EXIDEVICE_MEMORYCARDFOLDER:
+      index = 3;
+      break;
+    case ExpansionInterface::EXIDEVICE_GECKO:
+      index = 4;
+      break;
+    case ExpansionInterface::EXIDEVICE_AGP:
+      index = 5;
+      break;
+    case ExpansionInterface::EXIDEVICE_MIC:
+      index = 6;
+      break;
+    default:
+      break;
+    }
+
+    if (i < 2)
+    {
+      bool has_config = (index == 2 || index > 4);
```

Comment:
Same concern about readability here. It's not immediately obvious what index 2 or > 4 mean, and if the indexes ever need to change, it would be very easy to make mistakes, IMO.